File: src/main/java/io/github/dqualizer/dqlang/types/rqa/definition/stimulus/loadprofile/LoadPeak.java
package io.github.dqualizer.dqlang.types.rqa.definition.stimulus.loadprofile;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.symbolic.SymbolicValue;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.SuperBuilder;

@Getter
@ToString
@SuperBuilder
@NoArgsConstructor
public class LoadPeak extends LoadProfile {

    /**
     * Type identifier for JSON serialization.
     */
    public static final String TYPE_IDENTIFIER = "peak";

    @JsonProperty("peak_load")
    private SymbolicValue peakLoad;
    private SymbolicValue duration;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/rqa/definition/stimulus/loadprofile/LoadPeak.kt
package io.github.dqualizer.dqlang.types.rqa.definition.stimulus.loadprofile

import com.fasterxml.jackson.annotation.JsonProperty
import io.github.dqualizer.dqlang.types.rqa.definition.stimulus.symbolic.SymbolicValue

data class LoadPeak(
    @JsonProperty("base_load")
    override val baseLoad: SymbolicValue? = null,

    @JsonProperty("peak_load")
    val peakLoad: SymbolicValue? = null,

    val duration: SymbolicValue? = null
) : LoadProfile() {
    companion object {
        const val TYPE_IDENTIFIER: String = "peak"
    }
}
